resources/uco_design_document.md

```diff
@@ -181,7 +213,35 @@ By default the UUIDs used in object identifiers should be randomly generated UUI
+  * In some cases (where purpose-specific object properties may not be appropriate) this means defining, managing, tracking and processing many unnecessary object properties. Each new object property is a new predicate and therefore a new type of graph edge.
+  * In many cases (where purpose-specific object properties may not be appropriate) this leads to unnecessary and sometimes impractical churn on the UcoObject shapes affecting adopters.
+     * This is particularly true for new `UcoInherentCharacterizationThing` classes added to application subdomain ontologies built on UCO and to very specific usage communities and scenarios.
+         * In many of these cases the new `UcoInherentCharacterizationThing` classes may only be relevant to the application subdomain ontologies or very specific usage communities and scenarios. In these cases modifying the general shape for the `UcoObject` at the UCO level would be inappropriate.
+     * This is also true for defining particular subclasses of existing `UcoInherentCharacterizationThing` classes
+* In some cases (e.g., Identity) (where purpose-specific object properties may not be appropriate) this also makes it more complicated for particular adopters (e.g., application subdomain ontologies or even specific communities or organizations) to scope their adoption to specific `UcoInherentCharacterizationThing` class-based characteristics.
+  * The purpose-specific object property approach requires them to define their own specialized complex shapes (defining property shapes for all of the separate purpose-specific object properties) for the relevant UcoObjects.
+* In some cases (e.g., Location) (where purpose-specific object properties may not be appropriate) this also makes it unnecessarily more complicated for expressing the same basic concept characterization using different mechanisms (e.g., a location expressed utilizing different coordinate systems).
+* In some cases where specific context may not yet be understood and more general property association is necessary (e.g. duck typing), the purpose-specific object properties approach is impractical.
+* In some cases where differing levels of detail may be needed to characterize a given concept and handling it all through subclassing involves impractical complexity. This can be seen in the UCO approach for observable objects due to lessons learned in the Cyber Observable eXpression (CybOX) standardization effort which is the base origin for the UCO approach. CybOX found it highly detrimental to attempt to manage multiple levels of characterization detail through purpose-specific properties on complex subclass hierarchies (e.g., a plain file, a plain raster image file, a raster image file with EXIF data, etc.). Similar lessons were also learned in the Structured Threat Information eXpression (STIX) standardization effort.

```

Comment:
This paragraph does not finish explaining the drawback.  From a naïve read, a tree-structured class structure should work fine, especially with the straight-line lineage you illustrated.  Are you able to provide an example from the CybOX era where the impracticality was recognized, e.g. when the subclass hierarchy transitioned from tree to directed acyclic graph with multiple inheritance?